python_ta/transforms/type_inference_visitor.py

```diff
@@ -284,8 +284,10 @@ def get_call_signature(self, c, node: NodeNG) -> TypeResult:
         elif isinstance(c, _ForwardRef):
             class_type = c
             class_name = c.__forward_arg__
-        else:
+        elif isinstance(c, CallableMeta) or (hasattr(c, '__origin__') and c.__origin__ is Union):
```

Comment:
Why is `Union` permitted here?

Reply:
Overloaded functions pass in a Union of different possible function signatures. However, I'm now realizing this is too general, as it would then accept any Union, even if the elements were not function signatures, so I will adjust accordingly